To: vim_dev@googlegroups.com
Subject: Patch 7.3.367
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.367
Problem:    :wundo and :rundo use a wrong checksum.
Solution:   Include the last line when computing the hash. (Christian Brabandt)
Files:	    src/undo.c


*** ../vim-7.3.366/src/undo.c	2011-01-22 21:25:07.000000000 +0100
--- src/undo.c	2011-12-08 14:19:03.000000000 +0100
***************
*** 719,725 ****
      char_u		*p;
  
      sha256_start(&ctx);
!     for (lnum = 1; lnum < curbuf->b_ml.ml_line_count; ++lnum)
      {
  	p = ml_get(lnum);
  	sha256_update(&ctx, p, (UINT32_T)(STRLEN(p) + 1));
--- 719,725 ----
      char_u		*p;
  
      sha256_start(&ctx);
!     for (lnum = 1; lnum <= curbuf->b_ml.ml_line_count; ++lnum)
      {
  	p = ml_get(lnum);
  	sha256_update(&ctx, p, (UINT32_T)(STRLEN(p) + 1));
*** ../vim-7.3.366/src/version.c	2011-12-08 15:12:08.000000000 +0100
--- src/version.c	2011-12-08 15:13:32.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     367,
  /**/

-- 
hundred-and-one symptoms of being an internet addict:
247. You use www.switchboard.com instead of dialing 411 and 555-12-12
     for directory assistance.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
